File: src/main/java/eschool/courses/model/AcademicYear.java
package eschool.courses.model;

import javax.jdo.annotations.*;

@PersistenceCapable
public class AcademicYear {
	@PrimaryKey
	@Persistent(valueStrategy=IdGeneratorStrategy.INCREMENT)
	private long id;
	
	@Unique
	@Column(allowsNull="false")
	private String name;
	
	public AcademicYear() {}
	
	public AcademicYear(String name) {
		this.name = name;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public long getId() {
		return id;
	}
}

File: src/main/scala/eschool/courses/model/AcademicYear.scala
package eschool.courses.model

import javax.jdo.annotations._
import jdo.Id

class AcademicYear extends Id[Long] {
  @Unique
  @Column(allowsNull="false")
  private[this] var _name: String = _
  
  def this(name: String) = {
    this()
    _name = name
  }
  
  def name: String = _name
  def name_=(theName: String) { _name = theName }
}

object QAcademicYear 